File: rider/src/main/kotlin/com/jetbrains/resharper/plugins/unity/ideaInterop/fileTypes/shaderLab/ShaderLabLanguage.kt
package com.jetbrains.resharper.plugins.unity.ideaInterop.fileTypes.shaderLab

import com.jetbrains.resharper.ideaInterop.fileTypes.RiderLanguageBase

object ShaderLabLanguage : RiderLanguageBase("ShaderLab", "SHADERLAB") {
    override fun isCaseSensitive(): Boolean = false
}

File: resharper/src/resharper-unity/Psi/ShaderLab/ShaderLabLanguage.cs
using JetBrains.Annotations;
using JetBrains.ReSharper.Psi;

namespace JetBrains.ReSharper.Plugins.Unity.Psi.ShaderLab
{
    [LanguageDefinition(Name)]
    public class ShaderLabLanguage : KnownLanguage
    {
        public new const string Name = "SHADERLAB";

        [CanBeNull]
        public static readonly ShaderLabLanguage Instance = null;

        public ShaderLabLanguage()
            : base(Name, "ShaderLab")
        {
        }
    }
}